user-console/src/main/java/org/pentaho/mantle/client/solutionbrowser/SolutionBrowserPanel.java

```diff
@@ -837,4 +886,8 @@ private void createSchedule( final RepositoryFile repositoryFile ) {
   private native void createSchedule( final String repositoryFileId, final String repositoryFilePath )/*-{
     $wnd.pho.createSchedule( repositoryFileId, repositoryFilePath );
   }-*/;
+
+  private boolean isRepositoryPath( String path ){
```

Comment:
always try to DRY (Don't Repeat Yourself)

with GWT, we can't always re-use third party libraries (ie OSS and even pentaho code) due to the translation from java to javascript.

I would look into if we re-use existing pentaho logic that determines if a file is a "repository path". Yes the check is simple as you have it. I did a quick search and by no means exhaustive.

 look at FileUtils#isRootLevelPath - https://github.com/pentaho/pentaho-platform/blob/cfd04f738ace9437fc7133f4540fe7045114f5b0/extensions/src/main/java/org/pentaho/platform/web/http/api/resources/utils/FileUtils.java#L126
 
 This example may not be appropriate for your case.
 
 Regardless of implementation, please have unit tests for SolutionBrowserPanel#isRepositoryPath

Reply:
[GenericFilenameUtils ](https://github.com/pentaho/pentaho-commons-gwt-modules/blob/5431c20caf89844726d62f7145bd0f84752b2d3a/widgets/src/main/java/org/pentaho/gwt/widgets/client/genericfile/GenericFileNameUtils.java#L43) has exactly what we need, I am just having that compilation error on the @Null annotations... it would be nice if I can resolve this without changing that class at all

Reply:
Spoke with Duarte, if I add the annotation's dependency to the pom, I can compile the module, it is only failing in Super Dev Mode now for me, which is a shame, but not a blocker, as it's now just a dev tool problem not code problem...

So I will have that for this comment and [this one](https://github.com/pentaho/pentaho-platform/pull/5616#discussion_r1626131579)

Reply:
This SDM problem has been resolved via some dependency changes [here](https://github.com/pentaho/pentaho-platform/pull/5616/files#diff-b95754c647cdb0e7a2f6678afd4e031dedaccedaf020d7b2b5e43f5bdcddde46R54) and [in gwt-modules](https://github.com/pentaho/pentaho-commons-gwt-modules/pull/1039)